src/components/util/getBarChartOptions.ts

```diff
@@ -1,150 +1,175 @@
-import { ChartOptions } from 'chart.js';
 import { DataResponse, Measure } from '@embeddable.com/core';
+import { ChartDataset, ChartOptions } from 'chart.js';
+
 import formatValue from '../util/format';
 import { setYAxisStepSize } from './chartjs/common';
 import { Props } from './getStackedChartData';
 
+// We're adding a few properties to use when showing totals on the chart
+type ExtendedChartDataset = ChartDataset<'bar' | 'line'> & {
+  totals?: { [key: string]: { total: number; lastSegment: number | null } };
+  xAxisNames?: string[];
+};
+
 export default function getBarChartOptions({
-  results,
-  showLegend = false,
-  showLabels = false,
   displayHorizontally = false,
-  stacked = false,
-  stackMetrics = false,
-  displayAsPercentage = false,
-  yAxisTitle = '',
-  xAxisTitle = '',
   dps = undefined,
-  reverseXAxis = false,
-  metrics,
   lineMetrics,
   metric,
+  metrics,
+  results,
+  reverseXAxis = false,
+  secondAxisTitle = '',
+  segment,
+  showLabels = false,
+  showLegend = false,
   showSecondYAxis = false,
-  secondAxisTitle = ''
+  showTotals = false,
+  stackMetrics = false,
+  stacked = false,
+  xAxis,
+  xAxisTitle = '',
+  yAxisTitle = '',
+  displayAsPercentage = false,
 }: Partial<Props> & {
+  lineMetrics?: Measure[];
+  metric?: Measure;
+  metrics?: Measure[];
   results?: DataResponse;
-  stacked?: boolean;
-  stackMetrics?: boolean;
-  yAxisTitle?: string;
-  xAxisTitle?: string;
   reverseXAxis?: boolean;
-  metrics?: Measure[];
-  metric?: Measure;
-  showSecondYAxis?: boolean;
   secondAxisTitle?: string;
-  lineMetrics?: Measure[];
+  showSecondYAxis?: boolean;
+  stackMetrics?: boolean;
+  stacked?: boolean;
+  xAxisTitle?: string;
+  yAxisTitle?: string;
 }): ChartOptions<'bar' | 'line'> {
+  // Padding buffer for labels / totals as applicable
+  const bottom = 0;
+  let left = 0;
+  let right = 0;
+  let top = 0;
+  if (!stacked) {
+    if (displayHorizontally) {
+      right = showLabels ? 60 : 0;
+      left = showLabels ? 60 : 0;
+    } else {
+      top = showLabels ? 20 : 0;
+    }
+  } else {
+    if (displayHorizontally) {
+      right = showTotals ? 60 : 0;
+      left = showTotals ? 60 : 0;
+    } else {
+      top = showTotals ? 20 : 0;
+    }
+  }
+
```

Comment:
suggestion, instead of having this inside the component, you could create a function that returns the padding